spring-transaction/src/main/kotlin/org/jetbrains/exposed/spring/SpringTransactionManager.kt

```diff
@@ -115,6 +130,50 @@ class SpringTransactionManager(
                 addLogger(StdOutSqlLogger)
             }
         }
+
+        lateinit var con: Connection
+        try {
+            if (!trxObject.hasConnectionHolder() ||
+                trxObject.getConnectionHolder().isSynchronizedWithTransaction
+            ) {
+                trxObject.setConnectionHolder(
+                    ConnectionHolder(ExposedConnectionHandle(transaction))
+                )
+                trxObject.isNewConnectionHolder = true
+            }
+
+            trxObject.getConnectionHolder().setSynchronizedWithTransaction(true)
+            con = trxObject.getConnectionHolder().getConnection()
+
+            val previousIsolationLevel = DataSourceUtils.prepareConnectionForTransaction(con, definition)
+            trxObject.previousIsolationLevel = previousIsolationLevel
+            trxObject.isReadOnly = definition.isReadOnly
+
+            // Switch to manual commit if necessary. This is very expensive in some JDBC drivers,
+            // so we don't want to do it unnecessarily (for example if we've explicitly
+            // configured the connection pool to set it already).
+            if (con.autoCommit) {
+                trxObject.mustRestoreAutoCommit = true
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Switching JDBC Connection [" + con + "] to manual commit")
+                }
+                con.autoCommit = false
+            }
+
+            val timeout = determineTimeout(definition)
+            if (timeout != TransactionDefinition.TIMEOUT_DEFAULT) {
+                trxObject.getConnectionHolder().setTimeoutInSeconds(timeout)
+            }
+
+            // Bind the connection holder to the thread.
+            if (trxObject.isNewConnectionHolder){
+                TransactionSynchronizationManager.bindResource(dataSource, trxObject.getConnectionHolder())
+            }
+        } catch (ex: Throwable) {
+            DataSourceUtils.releaseConnection(con, dataSource)
+            trxObject.setConnectionHolder(null)
+            throw CannotCreateTransactionException("Could not open JDBC Connection for transaction", ex)
+        }
```

Comment:
This block is more or less lifted from `DataSourceTransactionManager`:

https://github.com/spring-projects/spring-framework/blob/main/spring-jdbc/src/main/java/org/springframework/jdbc/datasource/DataSourceTransactionManager.java#L261

Reply:
I intend to also check the `JpaTransactionManager`, which is an implementation in the same seat as the Exposed one, to see if there is anything to deduce from its similarities/differences to `DataSourceTransactionManager`

https://github.com/spring-projects/spring-framework/blob/main/spring-orm/src/main/java/org/springframework/orm/jpa/JpaTransactionManager.java